### PR TITLE
Add Java and database compatibility docs.

### PR DESCRIPTION
**Pull request by bame-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/ops/requirements.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+1. An x86-compatible system running a modern Unix, Windows, or MacOS operating system.
+2. Java 11 or greater.
+3. An RDBMS system,
+  
+  1. Either PostgreSQL 10.0 or greater.
+  2. Or Oracle Database 19.11 or greater.
+
+Daml is tested using the following specific dependencies in default installations.
+
+1. Opersting Systems:
```
Comment (cocreature) [inline docs/source/ops/requirements.rst:20]:
```suggestion
1. Operating Systems:
```

File: docs/source/ops/requirements.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  
+  1. Ubuntu 20.04
+  2. Windows Server 2016
+  3. MacOS 10.15 Catalina
+
+2. `Eclipse Adoptium <https://adoptium.net>`_ version 11 for Java.
```
Comment (cocreature) [inline docs/source/ops/requirements.rst:26]: not quite what we test with on CI (we get it from nix) but it’s an openjdk 11 so probably close enough

**Review by cocreature — APPROVED** [review]
Thanks!

File: docs/source/ops/requirements.rst (hunk 2 above)
Comment (garyverhaegen-da) [reply docs/source/ops/requirements.rst:26]: Adoptium is the name of the project, but their JVMs are named "Eclipse Temurin".

File: docs/source/ops/requirements.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _ops-ref_index:
+
+System Requirements
+===================
+
+Unless otherwise stated, all Daml runtime components require the following dependencies:
+
+1. An x86-compatible system running a modern Unix, Windows, or MacOS operating system.
```
Comment (garyverhaegen-da) [inline docs/source/ops/requirements.rst:11]: "Modern macOS" is not "x86-compatible" unless you count emulation. 😿

Do we actually (want to) support running production servers on macOS?

**Pull request merged**